src/sql/parts/notebook/notebook.component.ts

```diff
@@ -159,12 +171,23 @@ export class NotebookComponent extends AngularDisposable implements OnInit {
 		await model.requestModelLoad(this.notebookParams.isTrusted);
 		model.contentChanged((change) => this.handleContentChanged(change));
 		this._model = model;
+		this.updateToolbarComponents(this._model.trustedMode);
 		this._register(model);
 		this._modelRegisteredDeferred.resolve(this._model);
 		model.backgroundStartSession();
 		this._changeRef.detectChanges();
 	}
 
+	//Updates toolbar components
+	private updateToolbarComponents(isTrusted: boolean)
+	{
+		if(this._trustedAction)
```

Comment:
This is fine for now, but long-term we should make all these kind of special actions generic. E.g. we'll have contributions that can be called with a modelLoaded method that they can use for reading necessary values.